File: event-service/src/main/java/org/tggc/eventservice/mapper/EventMapper.java
package org.tggc.eventservice.mapper;

import org.mapstruct.Mapper;
import org.tggc.eventservice.dto.EventRq;
import org.tggc.eventservice.dto.EventRs;
import org.tggc.eventservice.model.Event;

import static org.mapstruct.MappingConstants.ComponentModel.SPRING;

@Mapper(componentModel = SPRING)
public interface EventMapper {

    EventRs toEventRs(Event event);

    Event toEvent(EventRq eventRq);
}

File: event-service/src/main/kotlin/org/tggc/eventservice/mapper/EventMapper.kt
package org.tggc.eventservice.mapper

import org.tggc.eventservice.dto.EventRq
import org.tggc.eventservice.dto.EventRs
import org.tggc.eventservice.model.Event

fun Event.toRs(): EventRs = EventRs(
    id = this.id,
    title = this.title,
    description = this.description,
    eventDate = this.eventDate,
    location = this.location,
    createdAt = this.createdAt,
    participants = this.participants
)

fun EventRq.toEventEntity(): Event = Event(
    title = this.title,
    description = this.description,
    eventDate = this.eventDate,
    location = this.location,
)